### PR TITLE
chore: defork cloud-provider-azure for 1.30

### PR DESCRIPTION
**Pull request by comtalyst**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Backport of https://github.com/kubernetes/autoscaler/pull/6947

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

Should be identical to the original PR.

#### Does this PR introduce a user-facing change?

```release-note
- Azure: from now on, users should refer to https://cloud-provider-azure.sigs.k8s.io/install/configs/ for configuration interface
- Azure: fixed an issue where environment variables were not being passed in when config file exists
- Azure: fixed an issue where some cloud provider configurations were not being validated when UseManagedIdentityExtension is set to true
- Azure: renamed several fields from config file, with old names are still acceptable and taking precedence: `useWorkloadIdentityExtension` to `useFederatedWorkloadIdentityExtension`, `vmssCacheTTL` to `vmssCacheTTLInSeconds`, `vmssVmsCacheTTL` to `vmssVirtualMachinesCacheTTLInSeconds`, `enableVmssFlex` to `enableVmssFlexNodes`
- Azure: renamed several environment variables, with old names are still acceptable and taking precedence: `ARM_USE_MANAGED_IDENTITY_EXTENSION` to `ARM_USE_FEDERATED_WORKLOAD_IDENTITY_EXTENSION`, `AZURE_VMSS_CACHE_TTL` to `AZURE_VMSS_CACHE_TTL_IN_SECONDS`, `AZURE_VMSS_VMS_CACHE_TTL` to `AZURE_VMSS_VMS_CACHE_TTL_IN_SECONDS`, `AZURE_ENABLE_VMSS_FLEX` to `AZURE_ENABLE_VMSS_FLEX_NODES`
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
```

### CODE REVIEW

File: cluster-autoscaler/go.mod [hunk 1]
```diff
@@ -55,6 +55,7 @@ require (
 	k8s.io/legacy-cloud-providers v0.0.0
 	k8s.io/utils v0.0.0-20231127182322-b307cd553661
 	sigs.k8s.io/cloud-provider-azure v1.29.4
```
Comment (tallaxes) [inline cluster-autoscaler/go.mod:57]: This should likely be 1.30.x? (And we probably need to give some thought to what it should be in master ...)

Comment (comtalyst) [reply cluster-autoscaler/go.mod:57]: (discussed offline) will be handled separately

**Pull request merged**